source/engine/graphics/techniques/ShadowMappingTechnique.cpp

```diff
@@ -108,8 +81,26 @@ namespace nc::graphics
         pipelineCreateInfo.setBasePipelineHandle(nullptr); // Graphics pipelines can be created by deriving from existing, similar pipelines. 
         pipelineCreateInfo.setBasePipelineIndex(-1); // Similarly, switching between pipelines from the same parent can be done.
 
-        m_pipeline = m_base->GetDevice().createGraphicsPipelineUnique(nullptr, pipelineCreateInfo).value;
-        m_base->GetDevice().destroyShaderModule(vertexShaderModule, nullptr);
+        m_pipeline = device.createGraphicsPipelineUnique(nullptr, pipelineCreateInfo).value;
+        device.destroyShaderModule(vertexShaderModule, nullptr);
+    }
+
+    ShadowMappingTechnique::~ShadowMappingTechnique() noexcept
```

Comment:
I see all of the techniques have destructors like this now. The vk handle types will free these automatically. I'm assuming the order of destruction matters here, but that should be handled by the ordering within the class, if possible.

Reply:
You're right, these have been around for a while.